app/screens/members_modal/members_modal.tsx

```diff
@@ -0,0 +1,332 @@
+            return;
+        }
+
+        setStartingButtonAction(false);
+    }, [onButtonTap, selectedIds, startingButtonAction]);
+
+    const handleGetProfiles = useCallback(debounce(() => {
+        if (next.current && !loading && !term && mounted.current) {
+            setLoading(true);
+            getProfiles().then(loadedProfiles);
+        }
+    }, 100), [getProfiles, loading, term]);
+
+    const search = useCallback(() => {
+        handleSearchUsers(term);
+    }, [handleSearchUsers, term]);
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    const onSearch = useCallback((text: string) => {
+        if (text) {
+            setTerm(text);
+            if (searchTimeoutId.current) {
+                clearTimeout(searchTimeoutId.current);
+            }
+
+            searchTimeoutId.current = setTimeout(() => {
+                handleSearchUsers(text);
+            }, General.SEARCH_TIMEOUT_MILLISECONDS);
+            return;
+        }
+
+        clearSearch();
+    }, [clearSearch, handleSearchUsers]);
+
+    const handleRemoveProfile = useCallback((id: string) => {
+        const newSelectedIds = Object.assign({}, selectedIds);
+
+        Reflect.deleteProperty(newSelectedIds, id);
+
+        setSelectedIds(newSelectedIds);
+    }, [selectedIds, setSelectedIds]);
+
+    const handleSelectProfile = useCallback((user: UserProfile) => {
+        if (selectedIds[user.id]) {
+            handleRemoveProfile(user.id);
+            return;
+        }
+
+        if (user.id === currentUserId) {
+            const selectedId = {[currentUserId]: true};
+            handleButtonTap(selectedId);
+            return;
+        }
+
+        const wasSelected = selectedIds[user.id];
+        if (!wasSelected && selectedCount >= maxSelectedUsers) {
+            return;
+        }
+
+        const newSelectedIds = Object.assign({}, selectedIds);
+        if (!wasSelected) {
+            newSelectedIds[user.id] = user;
+        }
+
+        setSelectedIds(newSelectedIds);
+        clearSearch();
+    }, [clearSearch, currentUserId, handleRemoveProfile, handleButtonTap, selectedIds, setSelectedIds]);
+
+    const data = useMemo(() => {
+        if (term) {
+            const exactMatches: UserProfile[] = [];
+            const filterByTerm = (p: UserProfile) => {
+                if (selectedCount > 0 && p.id === currentUserId) {
+                    return false;
+                }
+
+                if (p.username === term || p.username.startsWith(term)) {
+                    exactMatches.push(p);
+                    return false;
+                }
+
+                return true;
+            };
+
+            const results = filterProfilesMatchingTerm(searchResults, term).filter(filterByTerm);
+            return [...exactMatches, ...results];
+        }
+        return profiles;
+    }, [isSearch && selectedCount, isSearch && searchResults, profiles, term]);
```

Comment:
It is not clear when you look only at this function where `isSearch` comes into place. This is because `isSearch` means `Boolean(term)`. I would simply change the `if (term) {` to `if (isSearch)`.